include/net/nrf_cloud_cell_pos.h

```diff
@@ -35,15 +36,18 @@ struct nrf_cloud_cell_pos_result {
 	uint32_t unc;
 };
 
-/**@brief Request a cellular positioning query from nRF Cloud via MQTT.
+/**@brief Perform an nRF Cloud cellular positioning request via MQTT using
+ * the provided cell info.
  *
- * @param type        Type of cellular positioning request.
+ * @param cells_inf Pointer to cell info; if NULL, current (single) cell tower
+ *                  data will be requested from the modem and sent to the cloud.
  * @param request_loc If true, cloud will send location to the device.
  *                    If false, cloud will not send location to the device.
  * @retval 0 If successful.
  *           Otherwise, a (negative) error code is returned.
  */
-int nrf_cloud_cell_pos_request(enum nrf_cloud_cell_pos_type type, const bool request_loc);
+int nrf_cloud_cell_pos_request(const struct lte_lc_cells_info * const cells_inf,
+			       const bool request_loc);
```

Comment:
By doing this, the app is expected to use LTE_LC API to get cells_inf? This is not friendly design as not all app would use LTE_LC lib. For example the SLM app is completely AT command-based and can NOT add to use LTE_LC lib.

Reply:
the app does not need to use the lte_lc lib.  
it can obtain the cell information any way it chooses and add it to a `lte_lc_cells_info` struct.
i used `lte_lc_cells_info` since it is already defined and contains all the necessary items.
there are a couple items (`measurement_time` and `time_diff`) which are currently not used by cell pos,  I can mention them in the function comment. 

i don't see that much value in creating a new (almost identical) struct to use here, but if that is what everyone thinks is best, i will make the update.

Reply:
LTE LC uses %NCELLMEAS events to fill the data structure.
AT command based application can use %NBRGRSRP to get info for MCELL need.